torch_ipex/csrc/cpu/DevOPs.cpp

```diff
@@ -160,9 +164,27 @@ at::Tensor AtenIpexCPUDev::dil_convolution_overrideable(const at::Tensor & input
   DEBUG("AtenIpexCPUDev::convolution_overrideable\n");
   // NOTE: DO NOT always call contiguous. It may break lazy-reorder. Because contiguous will call reorder instantly.
   if (check_auto_dnnl()) {
+
+    // TODO: contiguous()
+    if (!check_tensor_own_shade_context(weight) ||
+        !cpu::ShadeDataContext::isDilTensor(weight) ||
+        cpu::ShadeDataContext::getDilTensor(weight).is_public_format()) {
+      auto packed_desc =
+          dil::convolution_forward::expected_weights_desc(
+              weight.sizes().vec(),
+              get_dil_data_type(weight.scalar_type()),
+              stride.vec(),
+              padding.vec(),
+              padding.vec(),
+              dilation.vec(),
+              groups,
+              dil::algorithm::convolution_direct);
+      bridge::reorderDilTensorGeneric(weight, packed_desc);
+    }
+
     return dil_convolution(
       input.is_contiguous() ? input : input.contiguous(),
-      weight.is_contiguous() ? weight : weight.contiguous(),
```

Comment:
We have an assumption: All the aten tensor should be contiguous if its buffer is the ideep tensor with block-format. So does the reordered weight tensor should always be contiguous?